src/components/Comments.js

```diff
@@ -0,0 +1,272 @@
+        const removeCommentOption = isCommentByLoggedInUser && (
+          <Glyphicon
+            glyph="trash"
+            onClick={() => {
+              this.deleteComment(comment.id);
+            }}
+            style={{ fontSize: '1em', color: '#003c7e', cursor: 'pointer' }}
+            title="Delete this comment"
+          />
+        );
+        commentList.push(
+          <div key={comment.id}>
+            <div
+              style={{ color: '#484848', backgroundColor: '#F5F5F5', borderRadius: '5px' }}
+              title={comment.timestamp}
+            >
+              <div style={{ padding: '0.5em' }}>
+                <b> <i> {comment.username} </i> </b>
+                <div style={{ float: 'right' }}>
+                  { removeCommentOption }
+                </div>
+                <br />
+                <span style={{ whiteSpace: 'pre-line' }}>
+                  {comment.text}
+                </span>
+              </div>
+            </div>
+            <br />
+          </div>,
+        );
+      });
+      this.setState({ commentList });
+    }
+  }
+
+  showCommentModal() {
+    this.setState({
+      showCommentModal: true,
+    });
+  }
+
+  hideCommentModal() {
+    this.setState({
+      showCommentModal: false,
+    });
+  }
+
+  captureComment(e: Event) {
+    if (e.target instanceof HTMLTextAreaElement) {
+      this.setState({
+        comment: e.target.value,
+      });
+    }
+  }
+
+  saveComment() {
+    const { docId } = this.props;
+    const { comment } = this.state;
+    const { searcher } = this.context;
+    const username = AuthUtils.getUserName(AuthUtils.getSavedUser());
+    const d = new Date();
+    const timestamp = `Posted on ${d.toLocaleDateString()} at ${d.toLocaleTimeString()}`;
+    const loggedDateTime = d.toISOString();
+    const id = username.concat(loggedDateTime);
+    // escape newline, carriage return and tab characters in the comment string
+    const formattedComment = comment.replace(/\n/g, '\\n').replace(/\r/g, '\\r').replace(/\t/g, '\\t');
+
+    const jsonDoc = `{ "fields" : { "comment_s" : [ "${formattedComment}" ], `
+      + `"docId_s" : [ "${docId}" ], "username_s" : [ "${username}" ], "date" : [ "${loggedDateTime}" ], `
+      + `"timestamp_s" : [ "${timestamp}" ], "table" : [ "comments" ] }, "id" : "${id}" }`;
+    searcher.search.addOrDeleteDocument(JSON.parse(jsonDoc), this.getComments);
+  }
+
+  clearComment() {
+    this.setState({
+      comment: '',
+    });
+  }
+
+  renderComments() {
+    const { commentList } = this.state;
+    const noCommentLabel = (
+      <div style={{ width: '100%', textAlign: 'center', padding: '1em', color: 'gray' }}>
+        <i> No Comments Available </i>
+      </div>
+    );
+    const comments = commentList.length > 0 ? commentList : noCommentLabel;
+    return comments;
+  }
+
+  renderCommentModal() {
+    const { showCommentModal, comment } = this.state;
+    const commentModal = (
```

Comment:
There's no reason to save this as a variable. Instead, just return this directly.
```
return (
  <Modal
    ...
  </Modal>
);
```